TidepoolKit/Model/Datum/TCGMSettingsDatum.swift

```diff
@@ -0,0 +1,240 @@
+            self.urgentLow = urgentLow
+            self.urgentLowPredicted = urgentLowPredicted
+            self.low = low
+            self.lowPredicted = lowPredicted
+            self.high = high
+            self.highPredicted = highPredicted
+            self.fall = fall
+            self.rise = rise
+            self.noData = noData
+            self.outOfRange = outOfRange
+        }
+
+        public struct DurationAlert: Codable, Equatable {
+            public enum Units: String, Codable {
+                case hours
+                case minutes
+                case seconds
+            }
+
+            public var enabled: Bool?
+            public var duration: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, duration: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.duration = duration
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct LevelAlert: Codable, Equatable {
+            public typealias Units = TBloodGlucose.Units
+
+            public var enabled: Bool?
+            public var level: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, level: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.level = level
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct RateAlert: Codable, Equatable {
+            public enum Units: String, Codable {
+                case milligramsPerDeciliterPerMinute = "mg/dL/minute"
+                case millimolesPerLiterPerMinute = "mmol/L/minute"
+            }
+
+            public var enabled: Bool?
+            public var rate: Double?
+            public var units: Units?
+            public var snooze: Snooze?
+
+            public init(enabled: Bool, rate: Double, units: Units, snooze: Snooze? = nil) {
+                self.enabled = enabled
+                self.rate = rate
+                self.units = units
+                self.snooze = snooze
+            }
+        }
+
+        public struct Snooze: Codable, Equatable {
+            public enum Units: String, Codable {
+                case hours
+                case minutes
+                case seconds
+            }
+
+            public var duration: Double?
+            public var units: Units?
+
+            public init(_ duration: Double, _ units: Units) {
+                self.duration = duration
+                self.units = units
+            }
+        }
+    }
+
+    public struct ScheduledAlert: Codable, Equatable {
+        public var name: String?
+        public var days: [String]?
+        public var start: Int?
+        public var end: Int?
+        public var alerts: Alerts?
```

Comment:
At least for the Dexcom implementation, the alert schedule can be enabled or disabled (as well as always sound enabled/disabled).

Reply:
It was promoted to the `Alerts ` struct above.